### DISCUSSION TITLE
Confused about presence and visible

### DISCUSSION THREAD

**Discussion by PankajSavaliya** [seleniumbase/SeleniumBase#2904] [General]
```
anchor_element = WebDriverWait(driver, 5).until(
                EC.presence_of_all_elements_located((By.XPATH, iframe_link_path))
            )
           self.find_visible_elements(selector, by="css selector", limit=0) 
            
```
```
anchor_elements = WebDriverWait(driver, 5).until(
                EC.element_to_be_clickable((By.XPATH, iframe_link_path))
            )
self.click_visible_elements(selector, by="css selector", limit=0, timeout=None)
```

is this correct or not as `presence` and `visible` mean different so confused.

please correct the syntax from raw to SB if I'm wrong. @mdmintz

**Comment by mdmintz** [comment]
`WebDriverWait` is from raw Selenium: https://github.com/SeleniumHQ/selenium
(Questions regarding the Selenium API should be asked there.)

SeleniumBase methods include automatic waiting, making `WebDriverWait` unnecessary and redundant.

**Reply by PankajSavaliya** [reply]
```
anchor_elements = WebDriverWait(driver, 5).until(
                EC.element_to_be_clickable((By.XPATH, iframe_link_path))
            )
            
anchor_elements = WebDriverWait(driver, 5).until(
                EC.element_to_be_clickable((By.XPATH, iframe_link_path))
            )

```

I know these methods are from Selenium, can you please suggest to me which method I should use from SB instead of these two, I'm unable to find them from [Method summary ](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/method_summary.md). I want to use the SB method instead of raw. @mdmintz

**Reply by mdmintz** [reply]
https://github.com/seleniumbase/SeleniumBase/blob/1c526f1c02a9c7f4ec538d1600829acff44aa05e/help_docs/method_summary.md?plain=1#L846

https://github.com/seleniumbase/SeleniumBase/blob/1c526f1c02a9c7f4ec538d1600829acff44aa05e/help_docs/method_summary.md?plain=1#L778

https://github.com/seleniumbase/SeleniumBase/blob/1c526f1c02a9c7f4ec538d1600829acff44aa05e/help_docs/method_summary.md?plain=1#L765

**Reply by PankajSavaliya** [reply]
this method only gives me one element, as my xpath has multiple elements.  so it has multiple elements.

my xpath selector sample
```
//iframe[@id='1'] | //iframe[@id='2'] | //iframe[@id='3']
``` 

@mdmintz

**Reply by mdmintz** [reply]
https://github.com/seleniumbase/SeleniumBase/blob/1c526f1c02a9c7f4ec538d1600829acff44aa05e/help_docs/method_summary.md?plain=1#L136

The selector can be both "css selector" and "xpath".

**Reply by PankajSavaliya** [reply]
for `element_to_be_clickable ` also? as I'm getting the error while using `find_elements ` for `element_to_be_clickable`

the error is 
`element click intercepted: Element is not clickable`

so before click I have to check clickable or not.

**Reply by mdmintz** [reply]
If you have element overlays, you would need to use: `self.js_click_all(selector)`

https://github.com/seleniumbase/SeleniumBase/blob/1c526f1c02a9c7f4ec538d1600829acff44aa05e/help_docs/method_summary.md?plain=1#L419

`ElementClickInterceptedException` isn't covered in waiting for an element to be clickable.
